src/custom/state/swap/trade.test.ts

```diff
@@ -0,0 +1,112 @@
+import { parseUnits } from '@ethersproject/units'
+import { getCanonicalMarket } from '@src/custom/utils/misc'
+import { PriceQuoteParams } from '@src/custom/utils/operator'
+import { basisPointsToPercent } from '@src/utils'
+import { ChainId, Fraction, JSBI, Pair, Route, Token, TokenAmount, Trade, TradeType, WETH } from '@uniswap/sdk'
+
+async function mockGetPriceQuote(params: PriceQuoteParams & { mockedPriceOut: string }) {
+  const { quoteToken, mockedPriceOut } = params
+  return Promise.resolve({
+    amount: mockedPriceOut,
+    token: quoteToken
+  })
+}
+
+const WETH_MAINNET = new Token(ChainId.MAINNET, WETH[1].address, 18)
+const DAI_MAINNET = new Token(ChainId.MAINNET, '0x6b175474e89094c44da98b954eedeac495271d0f', 18)
+
+const PAIR_WETH_DAI = new Pair(
+  new TokenAmount(WETH_MAINNET, JSBI.BigInt(1000000000)),
+  new TokenAmount(DAI_MAINNET, JSBI.BigInt(2000000000))
+)
+
+describe('Swap PRICE Quote test', () => {
+  describe('ExactIn: WETH/DAI_MAINNET', () => {
+    let inTrade, quote: any, extendedTradeIn: any
+
+    // 1 WETH SELL
+    const amountIn = parseUnits('1').toString()
+    const currencyIn = new TokenAmount(WETH_MAINNET, amountIn)
+
+    const MOCKED_PRICE_OUT = parseUnits('4273', DAI_MAINNET.decimals).toString()
+
+    beforeEach(async () => {
+      // make a new Trade object
+      inTrade = new Trade(new Route([PAIR_WETH_DAI], WETH_MAINNET), currencyIn, TradeType.EXACT_INPUT)
+
+      const { quoteToken, baseToken } = getCanonicalMarket({
+        sellToken: WETH_MAINNET.address,
+        buyToken: DAI_MAINNET.address,
+        kind: 'sell'
+      })
+
+      // get the WETH/DAI quote
+      // mock and return MOCKED_PRICE_OUT
+      quote = await mockGetPriceQuote({
+        // custom amount out we want as part of mock
+        mockedPriceOut: MOCKED_PRICE_OUT,
+        chainId: 1,
+        quoteToken,
+        baseToken,
+        amount: amountIn,
+        kind: 'sell'
+      })
+
+      extendedTradeIn = {
+        ...inTrade,
+        outputAmount: new TokenAmount(DAI_MAINNET, quote.amount),
+        maximumAmountIn: inTrade.maximumAmountIn,
+        minimumAmountOut: inTrade.minimumAmountOut
+      }
+    })
+
+    it('[Expect FAIL] ExactIN swap: 0.5% Slippage: <outputAmount * (1 - 0.005)>', async () => {
+      // we expect a slippage of (1 - slippage) e.g 1 - 0.005 = 0.995
```

Comment:
The test code is small, I like that!

I would highly encourage to follow the GIVEN/WHEN/THEN rule. It will make it easier to read, also sets clear expectation on what are you actually testing, which tbh, i'm not sure by reading the `it` description.

I guess here these tests are meant to test the `minimumAmountOut` function. This should be your `WHEN`.

Also, for the given, is best to **NOT DOING** calculations. You are adding code to test code. So you can easily make a mistake in the "expected result" you calculate. 

I think it works best to set some rounder numbers (like, for example, making the price 4000 instead of 4273), and adding some calculations only as comments. But expecting the exact result.